### ISSUE TITLE
(UX) Easier configuration of crash reporter

### ISSUE THREAD

**Issue by jamuc** [xing/crep#76]
Hey,

How about in addition to using command line params, you could also specify a JSON config file. Something like:
```
crashes:
    bundle_identifier: 
    build:
    ...... 
```

As a follow-up step, we could  use WebBrick, which is included in Ruby: https://en.wikipedia.org/wiki/WEBrick

We could allow the user editing the config file in the browser. With a save and validation of input fields.

Actually, could try and combine it with the apps controller and have a clickable/selectable list of identifiers. 

I might be overpromising here.

**Comment by BasThomas** [comment]
I think it would be interesting to at some point offer an option to pass a configuration via JSON, yes. The amount of arguments that have to be passed in at this point are already quite high, adding such an option would make sense to me.

Not sure if the browser configuration is something that would make sense. I think that pulls users away from the "simplicity" of the tool in its current state. What do you think?

**Comment by jamuc** [comment]
Yeah, I think the first step is to have a configuration file. Be it JSON, YAML (@q231950 favorite).
The other stuff is playing around. I would still want to keep it simple. That means there shouldn't be any extra steps, configration required by the user to get the tool working. On the contrary, it should reduce the amount of steps you have to take. If I have time this week I might give a stab at a prototype to show you guys.

**Comment by q231950** [comment]
@jamuc I agree with @BasThomas regarding simplicity. I think one topic that would be interesting is interactive mode. The `apps` command has all the infos which are required to create a _Crash Report_. The trick is to ask the user the right questions at the right time. I'm not sure though how to best approach _interactivity_ without cluttering the code. Probably there are some best practices around. Now that I am thinking about it, _interactive mode_ should be a separate issue - I'll search for it..

**Comment by q231950** [comment]
I created #79 that covers _Interactive Mode_.

**Comment by jamuc** [comment]
yeah, I like the interactive mode idea. Just a thought, you guys are the heavy users. What is the most common workflow for the tool? Could that be the default?

**Comment by q231950** [comment]
@jamuc I see 2 common use cases:

#1 a special report has to be created, I only know the git tag and bundle identifier, maybe rough date when it was published
#2 a regular _Crash Report_  where I want to have the report for the latest release for a known bundle identifier

In the first case it would be great if it just proposed build numbers and dates, in the second it should just take the bundle identifier and find the latest build itself. I think the most natural *default* would be it’s rawest form, no interactivity, no auto magic. I think that’s what I would expect it to behave like
